tests/unit/plugins/inventory/test_kubevirt.py

```diff
@@ -422,716 +501,561 @@ def test_ansible_connection_winrm(inventory, host_vars, client, vmi, expected):
-            "inventory_vmi_group",
-            "children_group_with_vmi_create_groups_option",
+            ["test", "namespace_default", "label_kubevirt_io_domain_test_domain"],
+            ["default-testvmi"],
+            {
+                "test": ["namespace_default"],
+                "namespace_default": ["default-testvmi"],
+                "label_kubevirt_io_domain_test_domain": ["default-testvmi"],
+            },
             True,
-            "complete_vmi_host_vars",
+            None,
+            COMPLETE_VMI_HOST_VARS,
             True,
             False,
         ),
         (
             {"vmis": [WINDOWS_VMI_1]},
             WINDOWS_VMI_1,
-            "inventory_groups",
-            "inventory_vmi_group",
-            "children_group_with_vmi",
+            ["test", "namespace_default"],
+            ["default-testvmi"],
+            {"test": ["namespace_default"], "namespace_default": ["default-testvmi"]},
             False,
-            "windows_vmi_host_vars",
+            None,
+            WINDOWS_VMI_HOST_VARS,
             True,
             True,
         ),
+        (
+            {
+                "vmis": [COMPLETE_VMI_WITH_NETWORK_NAME],
+                "services": [LOADBALANCER_SERVICE],
+            },
+            COMPLETE_VMI_WITH_NETWORK_NAME,
+            ["test", "namespace_default", "label_kubevirt_io_domain_test_domain"],
+            ["default-testvmi"],
+            {
+                "test": ["namespace_default"],
+                "namespace_default": ["default-testvmi"],
+                "label_kubevirt_io_domain_test_domain": ["default-testvmi"],
+            },
+            True,
+            "test-network",
+            COMPLETE_VMI_HOST_VARS_WITH_NETWORK,
+            True,
+            False,
+        ),
     ],
     indirect=["client"],
 )
-def test_get_vmis_for_namespace(mocker,
-                                request,
-                                inventory,
-                                vmi,
-                                host_vars,
-                                add_group,
-                                add_host,
-                                add_child,
-                                client,
-                                groups,
-                                vmi_group,
-                                child_group,
-                                create_groups,
-                                expected_host_vars,
-                                call_functions,
-                                windows,
-                                ):
-    set_ansible_host_and_port = mocker.patch.object(inventory, "set_ansible_host_and_port")
+def test_get_vmis_for_namespace(
+    mocker,
+    inventory,
+    vmi,
+    host_vars,
+    add_group,
+    add_host,
+    add_child,
+    client,
+    groups,
+    vmi_group,
+    child_group,
+    create_groups,
+    network_name,
+    expected_host_vars,
+    call_functions,
+    windows,
```

Comment:
Still torn about this, it just doesn't look right to me. Is there nothing we can do about it?

Reply:
To be addressed later on in a follow-up PR.